### ISSUE TITLE
What is the correct flow?

### ISSUE THREAD

**Issue by paulo-coutinho** [actions/cache#503]
Hi,

I read the docs but don't understand the correct flow to use it.

In my case i have:

1. checkout repo
2. build C++ dependencies (path: ~/.conan)
3. compile project and link with dependencies
4. package

Where i need put cache? Before step 2? After step 2? If i put before the "cache" will save everything from the folder? 

It is obscure in my mind.

Can anyone explain to me?

**Comment by dhadka** [comment]
Put the cache step where you want the dependencies restored (before step 2).  It will automatically add a save step at the end of your workflow (after step 4) to save the cache contents.

**Comment by dhadka** [comment]
Closing.  Please file a new issue or open a discussion if you have further questions.
